src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py

```diff
@@ -858,9 +866,17 @@ def __call__(
         is_strength_max = strength == 1.0
 
         # 5. Preprocess mask and image
-        mask, masked_image, init_image = prepare_mask_and_masked_image(
-            image, mask_image, height, width, return_image=True
-        )
+
+        init_image = self.image_processor.preprocess(image, height=height, width=width)
+        init_image = init_image.to(dtype=torch.float32)
+
+        mask = self.mask_processor.preprocess(mask_image, height=height, width=width)
+        # binarize mask
+        mask[mask < 0.5] = 0
+        mask[mask > 0.5] = 1
```

Comment:
```suggestion
        mask[mask < 0.5] = 0
        mask[mask > 0.5] = 1
```

shouldn't we do this in the preprocess function?